lib/awful/titlebar.lua

```diff
@@ -447,6 +459,51 @@ local titlebar = {
 -- @tparam gears.surface|string path
 -- @see gears.surface
 
+--- Add tooltip texts for titlebar buttons: active, inactive, plain
+-- @beautiful beautiful.titlebar_tooltip_messages
+-- @tparam table tooltips Key-Value syntax: key is button name, value is tooltip text.
+-- See default usage in variable `default_tooltip_messages` in titlebar component.
+-- @see awful.titlebar
+-- @usage -- Configure titlebar button tooltip texts in your theme.
+-- theme.titlebar_tooltip_messages = {
+--     close = "Close",
+--     minimize = "Minimize",
+--     maximized_active = "Unmaximize",
+--     maximized_inactive = "Maximize",
+--     floating_active = "Tiling",
+--     floating_inactive = "Floating",
+--     ontop_active = "NotOnTop",
+--     ontop_inactive = "OnTop",
+--     sticky_active = "NotSticky",
+--     sticky_inactive = "Sticky"
+-- }
+--
+
+--- Titlebar buttons tooltips, tooltip parameter `delay_show`
+-- @beautiful beautiful.titlebar_tooltip_delay_show
+-- @tparam integer delay See documentation for awful.tooltip `delay_show`
+-- @see awful.tooltip
```

Comment:
Yes, this is just link to the documentation of `delay_show` and others similar params which are already described in the `awful.tooltip` documentation part. 
If you want I can copy descriptions from `awful.tooltip` for all config parameters. Or I don't know if there exists some alternative solution to post the link inside the description part? Depends on you. 

In addition, it doesn't belong to this issue, but I saw that all next parameters inside this `titlebar.lua` have some wrong structure for description part, like this one `--- close_button_normal_hover.` This is wrongly shown in the documentation. Can I remove wrong symbols `_` to fix this doc page? What do you think?

Reply:
> If you want I can copy descriptions from awful.tooltip for all config parameters. Or I don't know if there exists some alternative solution to post the link inside the description part? Depends on you

@Elv13, your call. Should we have links to (basically here) function parameters? Like, the `delay_show` property here would link to the `delay_show` parameter inside the `awful.tooltip` constructor. I think it would be technically not so easy since it means we would have to put anchors on every possible function parameters.

I would be ok with "we copy-past the current brief description from the parameter".

> In addition, it doesn't belong to this issue, but I saw that all next parameters inside this titlebar.lua have some wrong structure for description part, like this one --- close_button_normal_hover. This is wrongly shown in the documentation. Can I remove wrong symbols _ to fix this doc page? What do you think?

Yeah, they definitely need some love.

These underscores are necessary here because that's the item name. It fails because the markdown tool from the doc generation tool interpret them as "I want italics here"... I don't know if putting the variable name in backticks would fix that or whether we need to trick them with backslashes... :thinking: 

Maybe better to do that in a separate PR to not flood this feature addition.

(I'm pretty sure this  markdown bug can be linked to https://github.com/awesomeWM/awesome/issues/3500)

Reply:
In the new doc, that would be semi doable. There is already a `<span>` tag in there I could convert to `<a>` without much effort. Plus, it's more organized now, so we could even highlight the row when the anchor is selected. But since that work is far from merged (and still have regressions, so it's not just me being lazy), I don't think this is actionable *now*. This PR can/should get merged before the new doc branch is finished.

![image](https://user-images.githubusercontent.com/340384/179374044-3f80971b-4eb1-4690-896f-4f6f1dda2626.png)

Reply:
Doc is improved.